server/server.go

```
@@ -284,18 +293,20 @@ func (s *IDPServer) oidcPrivateKey() (*signingKey, error) {
 			if err := json.Unmarshal(b, &sk); err == nil {
 				return &sk, nil
 			} else {
-				log.Printf("Error unmarshaling key: %v", err)
+				slog.Error("Error unmarshaling oidc key", slog.Any("error", err))
 			}
 		}
 		id, k := mustGenRSAKey(2048)
 		sk.Key = k
 		sk.Kid = id
 		b, err = json.Marshal(&sk)
 		if err != nil {
-			log.Fatalf("Error marshaling key: %v", err)
+			slog.Error("Error marshaling key", slog.Any("error", err))
+			os.Exit(1)
 		}
 		if err := os.WriteFile(keyPath, b, 0600); err != nil {
-			log.Fatalf("Error writing key: %v", err)
+			slog.Error("Error writing oid key", slog.Any("error", err))
+			os.Exit(1)
```

curious: why exit instead of returning the error?

---

Keeping behaviour same as previous `log.Fatalf`. Trying to keep changes to just logging in this PR.

---

The code surrounding does a lazy initialization of key material. The original's let it crash approach makes it easy to troubleshoot but it won't crash until the middle of a request. If its worth it we can refactor out the lazy so crashing/initialization happens during app startup rather than in the middle of a request.